util/imagetools/loader.go

```diff
@@ -0,0 +1,415 @@
+				var sbom json.RawMessage
+				_, err := remotes.FetchHandler(l.cache, fetcher)(ctx, layer)
+				if err != nil {
+					return err
+				}
+				dt, err := content.ReadBlob(ctx, l.cache, layer)
+				if err != nil {
+					return err
+				}
+				if err := json.Unmarshal(dt, &sbom); err != nil {
+					return err
+				}
+				mat.sbom = sbom
+			}
+		}
+	}
+	return nil
+}
+
+type provenance struct { // TODO: this is only a stub, to be refactored later
+	BuildSource     string            `json:",omitempty"`
+	BuildDefinition string            `json:",omitempty"`
+	BuildParameters map[string]string `json:",omitempty"`
+	Materials       []material
+}
+
+type material struct {
+	Type  string `json:",omitempty"`
+	Ref   string `json:",omitempty"`
+	Alias string `json:",omitempty"`
+	Pin   string `json:",omitempty"`
+}
+
+func (l *loader) scanProvenance(ctx context.Context, fetcher remotes.Fetcher, desc ocispec.Descriptor, mat *mat) error {
+	_, err := remotes.FetchHandler(l.cache, fetcher)(ctx, desc)
+	if err != nil {
+		return err
+	}
+	dt, err := content.ReadBlob(ctx, l.cache, desc)
+	if err != nil {
+		return err
+	}
+
+	var cfg binfotypes.ImageConfig
+	if err := json.Unmarshal(dt, &cfg); err != nil {
+		return err
+	}
+
+	if cfg.BuildInfo == "" {
+		return nil
+	}
+
+	dt, err = base64.StdEncoding.DecodeString(cfg.BuildInfo)
+	if err != nil {
+		return errors.Wrapf(err, "failed to decode buildinfo base64")
+	}
+
+	var bi binfotypes.BuildInfo
+	if err := json.Unmarshal(dt, &bi); err != nil {
+		return errors.Wrapf(err, "failed to decode buildinfo")
+	}
+
+	p := &provenance{}
+	defer func() {
+		mat.provenance = p
+	}()
+	if bs := bi.Attrs["context"]; bs != nil {
+		p.BuildSource = *bs
+	}
+
+	if fn := bi.Attrs["filename"]; fn != nil {
+		p.BuildDefinition = *fn
+	}
+
+	for key, val := range bi.Attrs {
+		if val == nil || !strings.HasPrefix(key, "build-arg:") {
+			continue
+		}
+		if p.BuildParameters == nil {
+			p.BuildParameters = make(map[string]string)
+		}
+		p.BuildParameters[strings.TrimPrefix(key, "build-arg:")] = *val
+	}
+
+	p.Materials = make([]material, len(bi.Sources))
+
+	for i, src := range bi.Sources {
+		// TODO: mark base image
+		p.Materials[i] = material{
+			Type:  string(src.Type),
+			Ref:   src.Ref,
+			Alias: src.Alias,
+			Pin:   src.Pin,
+		}
+	}
+
+	return nil
+}
+
+func (r *result) Configs() map[string]*ocispec.Image {
+	if len(r.mats) == 0 {
+		return nil
+	}
+	res := make(map[string]*ocispec.Image)
+	for p, m := range r.mats {
+		if m.config == nil {
+			continue
+		}
+		res[p] = m.config
+	}
+	return res
+}
+
+func (r *result) Provenances() map[string]*provenance {
+	if len(r.mats) == 0 {
+		return nil
+	}
+	res := make(map[string]*provenance)
+	for p, m := range r.mats {
+		if m.provenance == nil {
+			continue
+		}
+		res[p] = m.provenance
+	}
+	return res
+}
+
+func (r *result) SBOMs() map[string]json.RawMessage {
+	if len(r.mats) == 0 {
+		return nil
+	}
+	res := make(map[string]json.RawMessage)
```

Comment:
Hm, I wonder if we could do a best-effort parse and process this as a `map[string]interface{}`? That way we could do `{{ .SBOM.Name }}` or similar to access individual fields on the SBOM, while not tying the client directly to a single struct definition.